sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/DynamicJoinSelection.scala

```diff
@@ -89,13 +111,19 @@ object DynamicJoinSelection extends Rule[LogicalPlan] with JoinSelectionHelper {
           false
         }
 
-        val preferShuffleHash = preferShuffledHashJoin(stage.mapStats.get)
+        val preferShuffleHash = streamedPlan match {
+          case LogicalQueryStage(_, streamedStage: ShuffleQueryStageExec)
+            if streamedStage.isMaterialized && streamedStage.mapStats.isDefined =>
+            preferShuffledHashJoin(stage.mapStats.get, streamedStage.mapStats.get)
+
+          case _ => false
```

Comment:
this will be regression.. it prevents multi-join convert to shj, e.g.:

```
SortMergejoin 2
  SortMergeJoin 1
    Shuffle 1
    Shuffle 2
  Shuffe 3
```
The join 2 can not convert to shj even shuffle 3 is small

Reply:
Fix, could you help to review this PR again? 

Thanks